### ISSUE TITLE
Results per entity type missing entities

### ISSUE THREAD

**Issue by erotavlas** [explosion/spaCy#3993]
spacy version = 2.1.6

I trained my data to recognize 8 different entity types using 10 fold cross validation
When I step through my code and evaluate each of the 10 models, the `results['ents_per_type']` variable contains different number of entities for each fold.  Is this normal?  

I was expecting there to be consistently 8 entities for each result with some entities having precision recall and fscore of 0 if they didn't exist in that particular test set.  

It was causing me some problem calculating the final averages which is why I noticed it.

For example I can get a result like this from the overall results object containing `results['ents_f'], results['ents_p'] and results['ents_r']`
```
10-fold cross validation average score...
Average Recall: 85.15209283181693
Average Precision: 88.09937683644647
Average FScore: 86.56832860827618
```

But my per entity results from `results['ents_per_type']` are

```
10-fold cross validation average scores per entity type...
PERSON p:74.085 r:78.916 f:76.415
DATE p:94.367 r:99.136 f:96.671
PHONENUMBER p:81.55 r:92.067 f:86.16
ORG p:71.474 r:77.25 f:74.095
SPECIMENID p:84.682 r:87.2 f:85.799
ADDRESS p:85.798 r:85.798 f:85.798
EMAIL p:100.0 r:100.0 f:100.0
MRN p:43.75 r:43.75 f:43.75
```

And say I average up the precisions for all entities I get a value of approximately 79
This doesn't match with the overall average precision of 88

The way I calculated the per entity type values is

```
        result_per_ent = []
        
        #accumulate results for each fold into an array
        # for each model in the folds, load model and evaluate on the test set
        # append the results to the array
        # result_per_ent.append(results['ents_per_type'])

        # average result for k-fold cross validation - per entity type
        # first accumulate all the values into arrays
        ent_dict = {}
        for item in result_per_ent:
            for ent in item:
                if (ent not in ent_dict):
                    ent_dict[ent] = {'p': [], 'r': [], 'f': []}
                ent_dict[ent]['p'].append(item[ent]['p'])
                ent_dict[ent]['f'].append(item[ent]['f'])
                ent_dict[ent]['r'].append(item[ent]['r'])

        #average the results
        file.write("" + "\n")
        file.write(str(len(result_f)) + "-fold " + "cross validation average scores per entity type..." + "\n")
        for ent in ent_dict:
            file.write(str(ent) + " p: " + str(average(ent_dict[ent]['p'])) + " r: " + str(
                average(ent_dict[ent]['r'])) + " f: " + str(average(ent_dict[ent]['f'])) + "\n")
```

**Comment by honnibal** [comment]
I might be reading your code wrong, but I think the difference here is between macro average and micro average. If the entity types have different frequencies, then the unweighted average of the entity-type scores won't be the same as the score over the whole dataset. Imagine an entity type with only one example, that you get wrong, and an entity with 99 examples that you get right. Your Overall F-score will be 99%, but your micro-average will be close to 50%.

**Comment by adrianeboyd** [comment]
I think this is the bug fixed in #4052 where entity types with recall of 0 weren't included in ents_per_type.
